aplib.net-demo/Assets/Scripts/Configuration.cs

```diff
@@ -0,0 +1,277 @@
+        else if (ItemAmmoAmount < 0) ItemAmmoAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check ItemHealthAmount value.
+        if (ItemHealthAmount > GridVolume * _maxPercentage) ItemHealthAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (ItemHealthAmount < 0) ItemHealthAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check ItemRageAmount value.
+        if (ItemRageAmount > GridVolume * _maxPercentage) ItemRageAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (ItemRageAmount < 0) ItemRageAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check EnemyMeleeAmount value.
+        if (EnemyMeleeAmount > GridVolume * _maxPercentage) EnemyMeleeAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (EnemyMeleeAmount < 0) EnemyMeleeAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check EnemyRangedAmount value.
+        if (EnemyRangedAmount > GridVolume * _maxPercentage) EnemyRangedAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (EnemyRangedAmount < 0) EnemyRangedAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check if items and enemies don't go over grid vol * _maxPrecentage(0.7f)
+        CheckItemOverflow();
+        CheckEnemyOverflow();
+
+        // Print correct number in text field. Skip this when testing.
+        if (!_test)
+        {
+            inputFieldGridW.text = GridSizeX.ToString();
+            inputFieldGridH.text = GridSizeZ.ToString();
+            inputFieldRoom.text = RoomAmount.ToString();
+            inputFieldItemAmmo.text = ItemAmmoAmount.ToString();
+            inputFieldItemHealth.text = ItemHealthAmount.ToString();
+            inputFieldItemRage.text = ItemRageAmount.ToString();
+            inputFieldEnemyMelee.text = EnemyMeleeAmount.ToString();
+            inputFieldEnemyRanged.text = EnemyRangedAmount.ToString();
+
+            SetValues();
+        }
+    }
+
+    public void CheckItemOverflow()
+    {
+        TotalItemCount = ItemAmmoAmount + ItemHealthAmount + ItemRageAmount;
+        if (TotalItemCount > GridVolume * _maxPercentage)
+        {
+            int maxCount = Mathf.CeilToInt(GridVolume * _maxPercentage / 3);
+
+            TotalItemCount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+            ItemAmmoAmount = maxCount;
+            ItemHealthAmount = maxCount;
+            ItemRageAmount = maxCount;
+        }
+    }
+
+    public void CheckEnemyOverflow()
+    {
+        TotalEnemyCount = EnemyRangedAmount + EnemyMeleeAmount;
+        if (TotalEnemyCount > GridVolume * _maxPercentage)
+        {
+            int maxCount = Mathf.CeilToInt(GridVolume * _maxPercentage / 3);
+
+            TotalItemCount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+            EnemyMeleeAmount = maxCount;
+            EnemyRangedAmount = maxCount;
+        }
+    }
+
+    private void SetValues()
```

Comment:
What values does this set? Maybe give a better name and a `///<summary>`

Reply:
Good point.